### PR TITLE
MRG: Evoked stat example

### LINKED ISSUE #2268: ROI definition in picks - ENH suggestion
For the plot_topo functions it would be nice if the picks argument can take multiple channel groups Eg: picks = [[1,2,3],[4,5,6]]. The plots can be shown at the centroid of the channels within a group. This can reduce the need to plot all channels resulting in really small plots for data recorded with a high density channel system.

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
I'm proposing an example for ERP ROI plotting (a topic which comes up ever so often). It also shows how to run and display typical stats on ERP data.

Arguably,
Closes #2268

Better than #2602
Solves what remains of #2693

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
https://7545-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_stats_cluster_erp.html

**Comment by jona-sassenhagen** [comment]
@cbrnr @mmagnuski @agramfort thoughts? I know it's not perfect yet, but is the general direction useful?

**Comment by jona-sassenhagen** [comment]
I assume parts of it will be covered by #4859.

Also, this is the much-simplified code after the `evoked.plot_image(mask=...)` stuff got merged :)

Some things we could do to save code here:

- add title kwarg to `evoked.plot_image`
- make `evoked.plot_image` automatically display channel names on y axis (via an `('auto' | bool)` kwarg)
- add a helper to create ROI name to picks dicts for 10/20 layouts
- allow `evoked.plot_image` to take dicts for picks and axes (`epochs.plot_image` already does this)
- create a hack for/convince MPL to add an option for `plt.subplots` or `ImageGrid` to also share vlims across axes, like [here](https://matplotlib.org/examples/pylab_examples/multi_image.html)
- kwargs to the `to_data_frame` methods that allow returning values in a statsmodels/R friendly long format only for selected channels/time windows

**Comment by larsoner** [comment]
That sounds like a reasonable set of separate PRs

**Comment by jona-sassenhagen** [comment]
I can make PRs for the smaller ones very soon.

What's your take on:

>create a hack for/convince MPL to add an option for plt.subplots or ImageGrid to also share vlims across axes, like here

?

**Comment by larsoner** [comment]
Even if we convince MPL to do it, we'll need a backport probably. So why don't we implement it locally in `fixes.py` (or maybe a new `viz/fixes.py`) for now, and if someone is motivated, they can try to convince the MPL folks in parallel?

**Comment by jona-sassenhagen** [comment]
Makes sense.

**Comment by agramfort** [comment]
WIP or MRG?

**Comment by jona-sassenhagen** [comment]
WIP from my end ... insofar as I don't really have time right now.

**Comment by jona-sassenhagen** [comment]
https://7838-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_stats_cluster_erp.html

I guess it's good now.

File: tutorials/plot_stats_cluster_erp.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+"""
+===========================================================================
+Calculating and visualising statistical significance thresholds on EEG data
```
Comment (larsoner) [inline tutorials/plot_stats_cluster_erp.py]: Title is a bit too long, "Visualizing ..." is probably good enough

File: tutorials/plot_stats_cluster_erp.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+# Split up the data by the median length in letters via the attached metadata
+median_value = str(epochs.metadata[name].median())
+long = epochs[name + " > " + median_value]
+short = epochs[name + " < " + median_value]
+
+#############################################################################
+# If we have a specific point in space and time we wish to test, it can be
+# convenient to convert the data into Pandas Dataframe format. In this case,
+# the :class:`mne.Epochs` object has a convenient
+# :function:`mne.Epochs.to_data_frame` method, which returns a dataframe.
```
Comment (larsoner) [inline tutorials/plot_stats_cluster_erp.py]: Does not link properly, probably needs to be `:meth:`

File: tutorials/plot_stats_cluster_erp.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+# Visualize the results
+fig, axes = plt.subplots(nrows=3, figsize=(8, 8))
+vmax = np.abs(evoked.data).max() * 1e6
+
+# Iterate over ROIs and axes
+axes = axes.ravel().tolist()
+for roi_name, ax in zip(sorted(rois.keys()), axes):
+    picks = rois[roi_name]
+    evoked.plot_image(picks=picks, axes=ax, colorbar=False, show=False,
+                      clim=dict(eeg=(-vmax, vmax)), mask=significant_points)
```
Comment (larsoner) [inline tutorials/plot_stats_cluster_erp.py]: `time_unit='s'` to avoid warnings

**Comment by jona-sassenhagen** [comment]
https://7877-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_stats_cluster_erp.html

**Comment by agramfort** [comment]
example is really nice, however I find it overly complicated. An MNE user
to adapt this to his
data needs to be a

Python expert : for loops, dict comprehension
MNE expert :  know internals of layout
matplotlib : play with axes
Pandas : query and groupby

I am fine merging this but we should really not advertise this as an intro
example and
clearly present this as advanced

**Comment by jona-sassenhagen** [comment]
I hope we can simplify it soon when a few things done here have been turned into convenience functions (see discussion above).

**Comment by agramfort** [comment]
yes I agree. In the mean time let's just not advertise it as intro example.

as I said no objection to merge.

**Comment by larsoner** [comment]
Still needs:

> You need to add this to documentation.rst somewhere, probably under the Statistics heading

**Comment by jona-sassenhagen** [comment]
Like this?

**Comment by jona-sassenhagen** [comment]
Thanks guys!

**Comment by jona-sassenhagen** [comment]
@sappelhoff  also see here

**Pull request merged**